ntoskrnl/io/pnpmgr/pnpmgr.c

```diff
@@ -1760,6 +1765,81 @@ IopGetParentIdPrefix(PDEVICE_NODE DeviceNode,
     return Status;
 }
 
+static
+ULONG
+IopValidateID(IN PWCHAR Id,
+              IN SIZE_T MaxIdLen,
+              IN ULONG MaxSeparators,
+              IN BOOLEAN IsMultiSz)
+{
+   PWCHAR PtrSymbol;
+   ULONG_PTR StringEnd;
+   WCHAR Symbol;
+   ULONG SeparatorsCount = 0;
+   PWCHAR PtrPrevSymbol = NULL;
+
+   PAGED_CODE();
+
+   StringEnd = (ULONG_PTR)Id + MaxIdLen * sizeof(WCHAR);
+
+   for (PtrSymbol = Id; (ULONG_PTR)PtrSymbol < StringEnd; PtrSymbol++)
+   {
+      Symbol = *PtrSymbol;
+
+      if (Symbol == UNICODE_NULL)
+      {
+         if (!IsMultiSz || (PtrPrevSymbol && PtrSymbol == PtrPrevSymbol + 1))
+         {
+            if (MaxSeparators == SeparatorsCount ||
+                MaxSeparators == MAX_SEPARATORS_MULTISZ)
+            {
+               return ((ULONG_PTR)PtrSymbol - (ULONG_PTR)Id) / 2 + 1;
+            }
+
+            DPRINT("IopValidateID: SeparatorsCount - %d, MaxSeparators - %d\n",
+                   SeparatorsCount,
+                   MaxSeparators);
```

Comment:
You may (or not) put `MaxSeparators` just next to the other variable on the same line. Also, it may be better to use `%lu` instead of `%d`.

Reply:
When already ALL RoS developers will have a single standard? :)
Ok